### PR TITLE
qt6Packages.qtmqtt: init mdule at 6.5.0

### PR DESCRIPTION
**Pull request by Cynerd**
###### Description of changes

This module is not in mirrors for some reason.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/qt-6/srcs.nix [hunk 1]
```diff
@@ -115,6 +115,14 @@
       name = "qtmultimedia-everywhere-src-6.4.3.tar.xz";
     };
   };
+  qtmqtt = {
```
Comment (wegank) [inline pkgs/development/libraries/qt-6/srcs.nix:118]: `# DO NOT EDIT! This file is generated automatically.`

**Comment by wegank** [comment]
cc @NickCao

File: pkgs/development/libraries/qt-6/srcs.nix (hunk 1 above)
Comment (Cynerd) [reply pkgs/development/libraries/qt-6/srcs.nix]: Ups... I missed that.

I tweaked it but now there is a version specified that is not part of that srcs and I feel that it is going to be way too easy to introduce version mismatch in Qt this way.

Comment (NickCao) [reply pkgs/development/libraries/qt-6/srcs.nix]: Shouldn't the build fail on version mismatch? That way we can easily catch it with nixpkgs-review.

File: pkgs/development/libraries/qt-6/modules/qtmqtt.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+, pkg-config
+}:
+
+qtModule rec {
+  pname = "qtmqtt";
+  version = "6.4.3";
+  src = fetchurl {
+    url = "https://github.com/qt/qtmqtt/archive/refs/tags/v${version}.tar.gz";
+    sha256 = "P2CpIHVx0ya4eJf/UYFEitJEdCa/Qn7PD1CUCmY4TrE=";
+    name = "v${version}.tar.gz";
```
Comment (NickCao) [inline pkgs/development/libraries/qt-6/modules/qtmqtt.nix:16]: Isn't this the default?

File: pkgs/development/libraries/qt-6/modules/qtmqtt.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  };
+  qtInputs = [ qtbase ];
+  nativeBuildInputs = [ pkg-config ];
```
Comment (NickCao) [inline pkgs/development/libraries/qt-6/modules/qtmqtt.nix:19]: Maybe not required? There are no external dependencies.

Comment (Cynerd) [reply pkgs/development/libraries/qt-6/modules/qtmqtt.nix:19]: You are right.

File: pkgs/development/libraries/qt-6/modules/qtmqtt.nix (hunk 2 above)
Comment (Cynerd) [reply pkgs/development/libraries/qt-6/modules/qtmqtt.nix:16]: I copied it over when I moved it from `srcs` without thinking... yes it is not necessary.

File: pkgs/development/libraries/qt-6/modules/qtmqtt.nix [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{ qtModule
+, stdenv
+, lib
+, fetchurl
+, qtbase
+, udev
+, pkg-config
+}:
```
Comment (wegank) [inline pkgs/development/libraries/qt-6/modules/qtmqtt.nix:8]:
```suggestion
{ qtModule
, fetchurl
, qtbase
}:
```

Comment (Cynerd) [reply pkgs/development/libraries/qt-6/modules/qtmqtt.nix:8]: 👍

**Comment by NickCao** [comment]
Time to bump to 6.5.0: https://github.com/NixOS/nixpkgs/pull/224469

**Comment by Cynerd** [comment]
> Time to bump to 6.5.0: #224469

Done.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 226606` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>qt6.qtmqtt (qt6Packages.qtmqtt)</li>
    <li>qt6.qtmqtt.dev (qt6Packages.qtmqtt.dev)</li>
  </ul>
</details>

**Review by NickCao — APPROVED** [review]

**Pull request merged**